solutionbox/inception/datalab_solutions/inception/_util.py

```diff
@@ -184,3 +186,54 @@ def get_sources_from_dataset(p, dataset, mode):
     return p | 'Read source from BigQuery (%s)' % mode >> beam.io.Read(bq_source)
   else:
     raise ValueError('Invalid DataSet. Expect CsvDataSet or BigQueryDataSet')
+
+
+def decode_and_resize(image_str_tensor):
+  """Decodes jpeg string, resizes it and returns a uint8 tensor."""
+
+  # These constants are set by Inception v3's expectations.
+  height = 299
+  width = 299
+  channels = 3
+
+  image = tf.image.decode_jpeg(image_str_tensor, channels=channels)
+  # Note resize expects a batch_size, but tf_map supresses that index,
+  # thus we have to expand then squeeze.  Resize returns float32 in the
+  # range [0, uint8_max]
+  image = tf.expand_dims(image, 0)
+  image = tf.image.resize_bilinear(image, [height, width], align_corners=False)
+  image = tf.squeeze(image, squeeze_dims=[0])
+  image = tf.cast(image, dtype=tf.uint8)
+  return image
+
+
+def resize_image(image_str_tensor):
+  """Decodes jpeg string, resizes it and re-encode it to jpeg."""
+
+  image = decode_and_resize(image_str_tensor)
+  image = tf.image.encode_jpeg(image, quality=100)
```

Comment:
could decode_and_resize do
1) decode_jpeg
2) stuff
3) encode_jpeg

so then decode_and_resize can be renamed resize_image